### PR TITLE
[dev-v2.7 istio] Add condition to check PSP compatibility

### PR DESCRIPTION
**Pull request by doflamingo721**
Linked issue:
https://github.com/rancher/rancher/issues/39847

### CODE REVIEW

File: charts/rancher-istio/101.1.1+up1.15.3/charts/tracing/templates/psp.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "tracing.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Values.provider }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
```
Comment (stormqueen1990) [inline charts/rancher-istio/101.1.1+up1.15.3/charts/tracing/templates/psp.yaml:10]: I think this `ServiceAccount` can be left out of the condition, as it is a good idea to have a service account specific for a deployment and it is currently used by the tracing deployment.

@aiyengar2 do you have any strong opinions on this?

File: charts/rancher-istio/101.1.1+up1.15.3/templates/clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  - '*'
+{{- if and .Values.global.rbac.pspEnabled (.Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy") }}
+- apiGroups:
+  - policy
+  resourceNames:
+  - istio-installer
+  resources:
+  - podsecuritypolicies
+  verbs:
+  - use
```
Comment (stormqueen1990) [inline charts/rancher-istio/101.1.1+up1.15.3/templates/clusterrole.yaml:133]: Is an `{{- end }}` missing here perhaps?

File: packages/rancher-istio/1.15/rancher-istio/charts/templates/clusterrole.yaml [hunk 3]
```diff
@@ -122,6 +122,7 @@ rules:
   - serviceaccounts
   verbs:
   - '*'
+{{- if and .Values.global.rbac.pspEnabled (.Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy") }}
```
Comment (stormqueen1990) [inline packages/rancher-istio/1.15/rancher-istio/charts/templates/clusterrole.yaml:125]: I think this is missing a matching `{{- end }}`.

**Review by stormqueen1990 — COMMENTED** [review]
Hi there, @doflamingo721!

I added some comments for things I saw. Also saw that some files in the `packages` folder seem to be missing a matching `{{- end }}` for the `{{- if ... }}` clause.

File: charts/rancher-istio/101.1.1+up1.15.3/templates/clusterrole.yaml (hunk 2 above)
Comment (doflamingo721) [reply charts/rancher-istio/101.1.1+up1.15.3/templates/clusterrole.yaml:133]: Oh I missed it. I have made the changes could you please review it again?

File: packages/rancher-istio/1.15/rancher-istio/charts/templates/clusterrole.yaml (hunk 3 above)
Comment (doflamingo721) [reply packages/rancher-istio/1.15/rancher-istio/charts/templates/clusterrole.yaml:125]: Added the missing `end` clause.

**Comment by doflamingo721** [comment]
@stormqueen1990 I somehow missed adding `end` for the `if` clause. I have updated the PR can you please review it again?

**Review by stormqueen1990 — COMMENTED** [review]
Hi there, @doflamingo721!

I think it's only the service account concern that's left (I believe that will break the deployment the way it is right now).

File: charts/rancher-istio/101.1.1+up1.15.3/charts/tracing/templates/psp.yaml (hunk 1 above)
Comment (aiyengar2) [reply charts/rancher-istio/101.1.1+up1.15.3/charts/tracing/templates/psp.yaml:10]: That sounds fine to me! Let’s remove it from the conditional

**Comment by doflamingo721** [comment]
@stormqueen1990 @aiyengar2  I have removed the service account from the condition. Can you please review it again? TIA!

**Review by stormqueen1990 — APPROVED** [review]
LGTM!

**Review by aiyengar2 — APPROVED** [review]

**Comment by prachidamle** [comment]
The fixes for this Istio ticket need to be tested after Monitoring PSP changes are merged in https://github.com/rancher/rancher/issues/39846 as discussed in offline conversation with @mitulshah-suse @doflamingo721

**Comment by doflamingo721** [comment]
@prachidamle Merging the PR. I did some sanity testing around the charts and everything is working as expected.

**Pull request merged**